src/openthaigpt_pretraining/cli_tools/spm_trainer.py

```diff
@@ -0,0 +1,103 @@
+import os
+import argparse
+import regex as re
+from tqdm import tqdm
+from multiprocessing import Pool
+
+import sentencepiece as spm
+import pandas as pd
+from datasets import load_dataset, Dataset
+from tokenizers import SentencePieceUnigramTokenizer
+
+from openthaigpt_pretraining.utils.data_cleaning import clean_text, filter_out_line
+
+LINES_DELIMETER = "\n"
+
+
+def prepare_datasets(texts: str) -> dict:
+    preapared_texts = []
+    for text in texts['text']:  # for every doc
```

Comment:
Why we have this? I don’t think we should split new line when training generative model.

Reply:
OK remove it